tests/test_files.py

```diff
@@ -235,9 +235,24 @@ def globs_to_regex_params(
         ),
         globs_to_regex_params(
             ["*/foo"], case_insensitive=False, partial=True,
-            matches=["abc/foo/hi.py", "foo/hi.py"],
+            matches=["abc/foo/hi.py", "foo/hi.py", "abc/def/foo/hi.py"],
             nomatches=["abc/xfoo/hi.py"],
         ),
+        globs_to_regex_params(
+            ["*c/foo"], case_insensitive=False, partial=True,
+            matches=["abc/foo/hi.py"],
```

Comment:
Looks like this will also work:
```suggestion
            matches=["abc/foo/hi.py", "stuff/abc/foo/hi.py"],
```

Reply:
Oh wait, I see that a similar case is already in nomatches.. Let me reread the new doc explanation.

Reply:
It doesn't match. The next line includes `nomatches=["def/abc/foo/hi.py"]`

Reply:
Yeah, I saw that after writing the comment.